src/azure-cli-core/azure/cli/core/auth/identity.py

```diff
@@ -405,3 +405,39 @@ def get_environment_credential():
         getenv(AZURE_TENANT_ID))
     credentials = ServicePrincipalCredential(sp_auth, authority=authority)
     return credentials
+
+
+def _build_msal_client_credential(service_principal_auth):
```

Comment:
Using a `ServicePrincipalAuth` is still a little bit redundant and requires 2 conversions:

```
SP entry dict -> ServicePrincipalAuth object -> msal client_credential argument
```

We may want to simplify this process and convert `SP entry dict` directly to `msal client_credential argument`.